### PR TITLE
Add uv dependencies

### PR DESCRIPTION
**Pull request by gCaglia**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.6.5
```
Comment (una-gal) [inline .pre-commit-config.yaml]: v 0.6.9 is available since last week. Do you want to keep the 0.6.5 version?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    hooks:
+      # Fix imports
+      - id: ruff
+        args: ["--fix", "--select", "I"]
+      # Linter (with fix)
+      - id: ruff
+        args: ["--fix"]
+      # Formatter
+      - id: ruff-format
+        types_or: ["python"]
```
Comment (una-gal) [inline .pre-commit-config.yaml:13]: Do we want to let it run on Jupyter notebooks or not?

File: README.md [hunk 3]
```diff
@@ -6,8 +6,15 @@
 Describe why your topic is important and what you want to share with your audience
 
 ## Requirements
-Do not forget to indicate Python version and any other tools
-+ add requirements.txt or conda.yml or docker image or Binder/Google Collab link
+PyLadies encourage the usage of [uv](https://astral.sh/blog/uv) for dependency management. You can get uv by running the following command in your shell:
```
Comment (una-gal) [inline README.md]: I would rather add uv and ruff pre-commit here as a hard requirement with direct links to how to install them from the source docs.

File: README.md [hunk 4]
```diff
@@ -6,8 +6,15 @@
...
+```bash
+curl -LsSf https://astral.sh/uv/install.sh | sh
+```
+To get started, open the `pyproject.toml` file and set the required python version. The pre-selected version 3.8 is generally a safe choice for most use cases.
+
+After you have specified the python version, you can create a virtual environemnt with `uv venv` and add packages with `uv add <package>`. Before the workshop, you can generate a requirements.txt file by running `uv export > requirements.txt`.
+
+If you are not familiar with uv or are in a hurry, feel free to work with your familiar dependency management system and the PyLadies will handle the `pyproject.toml` for you.
```
Comment (una-gal) [inline README.md]: Please move the rest of the info (lines 13-17 to the Usage)

Comment (una-gal) [reply README.md]: Do we really want to handle pyproject.toml for them?

File: README.md [hunk 5]
```diff
@@ -18,3 +25,26 @@ Re-watch [this YouTube stream](link)
...
+
+To run this project on Google Colab, follow the following instructions:
+1. Visit [Google Colab](https://colab.research.google.com/)
+2. In the top left corner select "File" &#8594; "Open Notebook"
+3. Under "GitHub", enter the URL of the repo of this workshop
+4. Select one of the Notebooks within the repo.
+5. At the tob of the notebook, runthe following code:
+```bash
+!cd ./name-of-repo
+!pip install -r requirements.txt
```
Comment (una-gal) [inline README.md:36]: Would it be valid in case of pyproject.toml usage?

**Review by una-gal — CHANGES_REQUESTED** [review]
Added structural comments to the Readme, a version question to pre-commit

File: .pre-commit-config.yaml (hunk 1 above)
Comment (gCaglia) [reply .pre-commit-config.yaml]: I am updating it to the latest `v0.7.3`.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (gCaglia) [reply .pre-commit-config.yaml:13]: I will actually say let's not. 
We often have jupyter notebooks that are meant to be filled in with code, which from the perspective of ruff will make them `broken`, which could prevent commits. People could of course use `--no-verify` if we want to try and format notebooks.

File: README.md (hunk 3 above)
Comment (gCaglia) [reply README.md]: I adjusted the wording here and added additional links.

File: README.md (hunk 4 above)
Comment (gCaglia) [reply README.md]: I moved the section to the Usage section, overwriting what's currently there. 

Regarding the `pyproject.toml` file, I wouldn't say we "want" to handle it for them, but I was thinking to give the option if people are not familiar with it.
But then again, it's so simple, that I removed that part for now.

File: README.md (hunk 5 above)
Comment (gCaglia) [reply README.md:36]: I added a mention in `Usage` to indicate that the `requirements.txt` file needs to be created for GoogleColab ;)

**Review by una-gal — APPROVED** [review]

**Pull request merged**